src/journal/journald-server.c

```diff
@@ -1378,7 +1378,7 @@ int server_process_datagram(sd_event_source *es, int fd, uint32_t revents, void
 
         if (fd == s->syslog_fd) {
                 if (n > 0 && n_fds == 0)
-                        server_process_syslog_message(s, strstrip(s->buffer), ucred, tv, label, label_len);
+                        server_process_syslog_message(s, s->buffer + strspn(s->buffer, WHITESPACE), ucred, tv, label, label_len);
```

Comment:
I am bit confused. You suggest that we should always forward message to syslog _as is_. But @evverx's comment implies we should also store NUL bytes our self (as he checks journalctl output and not /var/log/messages)? I would strongly prefer storing whatever data we get over syslog socket, but since that is not happening then I'd prefer your suggestion over what @evverx wants.